src/config/license.rs

```diff
@@ -156,32 +156,33 @@ impl Config {
         };
 
         let (end_year, start_year) = if self.use_dynamic_year_ranges {
-            let now_date = Local::now().format("%a %b %d %T %Y %z").to_string();
+            let now_date = Local::now().format("%Y").to_string();
             let dates = get_git_dates_for_file(filename);
-            let (last_updated_date, created_date) = match &dates[..] {
-                [first_date, .., last_date] => (first_date, last_date),
-                [first_date] => (first_date, first_date),
-                _ => {
-                    debug!("Did not get any dates from git for file: {}", filename);
-                    (&now_date, &now_date)
-                }
-            };
-
             // Git formats the dates such that we get "Wed May 29 04:54:58 2024 +0100" we only care
             // about the 4th "field" which is the year.
-            let created_year = created_date
-                .split(' ')
-                .nth(4)
-                .expect("Unable to parse created year!");
-            let last_updated_year = last_updated_date
-                .split(' ')
-                .nth(4)
-                .expect("Unable to parse last updated year!");
+            let dates: Vec<_> = dates
+                .iter()
+                .map(|date| date.split(' ').nth(4).expect("Unable to parse year!"))
+                .collect();
+
+            let (last_updated_date, created_date) = if dates.is_empty() {
+                debug!("Did not get any dates from git for file: {}", filename);
+                (now_date.clone(), now_date)
+            } else {
+                (
+                    dates
+                        .iter()
+                        .max()
+                        .expect("Unable to parse last updated year!")
```

Comment:
These expect messages are now out of date we aren't parsing here.

Something like `"Unable to determine last updated date!"` would work